src/plots/cartesian/axes.js

```diff
@@ -3783,7 +3783,7 @@ axes.drawLabels = function(gd, ax, opts) {
                                 q > ax['_visibleLabelMin_' + anchorAx._id]
                             ) {
                                 t.style('display', 'none'); // hidden
-                            } else if(e.K === 'tick' && !idx) {
+                            } else if(e.K === 'tick' && !idx && t.style('display') != 'none') {
```

Comment:
There are some known CI failures that @emilykl is working on (re: downloading fonts) but one real failure. This line is [triggering a custom error](https://app.circleci.com/pipelines/github/plotly/plotly.js/11934/workflows/4960f3b1-0aec-48c2-9858-87e2da6c990d/jobs/264117) we wrote just for our test suite to catch a problem that's very hard to cover explicitly in tests:
```
Error: d3 selection.style called as getter: disallowed outside event handlers as it can fail for unattached elements. Use node.style.attribute instead.
```

Reply:
Right, thanks

Reply:
Also removed the overlooked console debug print

Reply:
Hey sorry, based on the error message "can fail for unattached elements" I think `window.getComputedStyle` is exactly the issue: that'll take into account whether the element is in the DOM and not hidden by an ancestor - which we don't want to do, we want to allow graphs to be rendered out-of-DOM and then inserted later. Please switch to `t.node().style.display`, also let's use `!==` rather than `!=`.

Reply:
Ah sorry, not yet familiar with this stuff